syft/frameworks/torch/mpc/falcon/falcon.py

```diff
@@ -23,3 +28,23 @@ def conv2d(filters: syft.ReplicatedSharingTensor, image, padding=0):
     if is_wrapper:
         result = result.wrap()
     return result
+
+
+def private_compare(x_bit_sh, r_bit, beta_b, m):
+    # 2)
+    u = FalconHelper.determine_sign(x_bit_sh - r_bit, beta_b)
+    # 3)
+    w = FalconHelper.xor(x_bit_sh, r_bit)
+    # 4)
+    wc = w.flip(-1).cumsum(-1).flip(-1) - w
+    c = u + wc + 1
+    # 6)
+    # TODO: Figure out how to do prod() privately?
```

Comment:
what seem to be the issue with prod()? 
btw, some hints can be taken from the source code used in the paper: 
https://github.com/snwagh/falcon-public/blob/7556f911baa52b320ff76852deadf20036d64b51/src/Functionalities.cpp#L1238

Reply:
Hmm..I think the problem here is that the reconstruction happens on the orchestrator and then we do the ```prod```.
And we should do "prod" (while it is still secretly shared) and then "reconstruction"?

Reply:
What George said.

Reply:
I understand that - but I am asking - what is the issue with doing prod on shares ?

Reply:
Yeah that's the thing. I'm not doing it on the shares but on the reconstructed value.

I need to figure out how to do it in on the shares.